python/lsst/analysis/tools/tasks/sourceObjectTableAnalysis.py

```diff
@@ -98,35 +241,147 @@ class SourceObjectTableAnalysisConfig(
+        with Matcher(isolatedSources["coord_ra"], isolatedSources["coord_dec"]) as m:
+            idx, i1, i2, d = m.query_radius(
+                properMotionCatalog["starX"], properMotionCatalog["starY"], 0.2 / 3600, return_indices=True
+            )
+        pmDf = properMotionCatalog.rename(
+            columns={
+                "starX": "ra",
+                "starY": "dec",
+                "starPMx": "pmRA",
+                "starPMy": "pmDec",
+                "starParallax": "parallax",
+            }
+        )
+        pmDf.insert(5, "MJD", sourceMjd)
+        raCorrection, decCorrection = calculate_apparent_motion(pmDf.iloc[i2], targetEpochs[i1])
+        isolatedSources["coord_ra"][i1] -= raCorrection.value
+        isolatedSources["coord_dec"][i1] -= decCorrection.value
+        return isolatedSources
 
+    def prepareAssociatedSources(
+        self, visit, data, associatedSourceRefs, refCats, visitTable, properMotionCatalog
+    ):
+        """Match isolated sources with reference objects and shift the sources
+        to the object epochs if the properMotionCatalog is not None.
+
+        Parameters
+        ----------
+        visit : `int`
+            Identifier of the visit corresponding to the data.
+        data : `astropy.table.Table`
+            Catalog of sources.
+        associatedSourceRefs : `list` [`DeferredDatasetHandle`]
+            Handle for the catalogs of isolated sources. There will be multiple
+            if the visit overlaps with multiple tracts.
+        refCats : `list` [`pd.DataFrame`]
+            Catalog of objects with which the sources will be compared.
+        visitTable : `pd.DataFrame`
+            Catalog containing the epoch for the visit corresponding to the
+            isolatedSources.
+        properMotionCatalog : `pd.DataFrame`
+            Catalog with proper motion and parallax information.
+        """
         isolatedSources = []
-        for associatedSourcesRef in inputs["associatedSources"]:
-            associatedSources = associatedSourcesRef.get(parameters={"columns": ["visit", "source_row"]})
+        for associatedSourceRef in associatedSourceRefs:
+            associatedSources = associatedSourceRef.get(parameters={"columns": ["visit", "source_row"]})
             visit_sources = associatedSources[associatedSources["visit"] == visit]
             isolatedSources.append(data[visit_sources["source_row"]])
         isolatedSources = vstack(isolatedSources)
 
         if len(isolatedSources) == 0:
-            raise NoWorkFound(f"No isolated sources found for visit {visit}")
+            raise pipeBase.NoWorkFound(f"No isolated sources found for visit {visit}")
+
+        with Matcher(isolatedSources["coord_ra"], isolatedSources["coord_dec"]) as m:
+            idx, isolatedMatchIndices, refMatchIndices, dists = m.query_radius(
+                refCats[self.config.ra_column].values,
+                refCats[self.config.dec_column].values,
+                1 / 3600.0,
```

Comment:
Why is this 1" and the others were 0.2"? Again, a magic number should at least be a function kwarg, and probably a config field.

Reply:
The others were matching two things that have the same provenance, and thus should have very similar positions, whereas this one is matching sources to objects. I made this a config field.